### PR TITLE
Replace feature flag with index setting and enable TSDB encoding and decoding

### PR DESCRIPTION
**Pull request by salvatore-campagna**
The index setting controls if the TSDB codec is enabled or not.
The codec applies a set of encoding schemes to gauges and counter
fields in time series indices including delta encoding, gcd encoding
offset encoding and so on. Encoding of values requiring less than 32
bits is vectorized.

Here we also enable encoding and decoding by default.

### CODE REVIEW

**Comment by salvatore-campagna** [comment]
I will include some Javadoc explaining the encoding/decoding schemes involved.

File: server/src/test/java/org/elasticsearch/index/codec/PerFieldMapperCodecTests.java [hunk 1]
```diff
@@ -49,6 +49,100 @@ public void testUseBloomFilterWithTimestampFieldEnabled_disableBloomFilter() thr
         );
     }
 
+    public void testUseES87TSDBEncodingForTimestampField() throws IOException {
```
Comment (martijnvg) [inline server/src/test/java/org/elasticsearch/index/codec/PerFieldMapperCodecTests.java:52]: Maybe also add a test that uses `index.time_series.es87tsdb_codec.enabled` setting

**Review by martijnvg — COMMENTED** [review]
This looks good.

> I will include some Javadoc explaining the encoding/decoding schemes involved.

This can also be added in a follow up change.

**Review by martijnvg — APPROVED** [review]
LGTM

**Pull request merged**